keps/sig-api-machinery/4977-dedicated-priority-level-for-event-requests/README.md

```diff
@@ -0,0 +1,970 @@
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Many features have been developed and use cases surfaced since the introduction
+of [KEP-1040]. These prompt us to revise APF default configuration to better
+make use of APF for better performance and reliability of API Server. One thing
+we have learned is that Event requests are plentiful in upscaling but they are
+resilient to unavailability of API Server. So we can deprioritize them when API
+Server is under stress.
+
+[KEP-1040]: https://github.com/kubernetes/enhancements/blob/97713189b3107b41c4c19505d04aa7ef22df063b/keps/sig-api-machinery/1040-priority-and-fairness/README.md
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+-   Better overload protection in perspective of whole Kubernetes control plane
+    instead of just API Server alone.
+-   More performant in cluster upscaling (e.g. increasing node count, deploying
+    pods)
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+-   There will be no fundamental changes in APF's own implementation
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+The APF feature provides flow control to API Server by introducing a limit on
+how much a priority level can consume the concurrency. In addition to that,
+borrowing is also instroduced to better utilize spare concurrency. The limits on
+borrowing are two-sided: a given priority level has a limit on how much it may
+borrow and a limit on how much may be borrowed from it. Worth mentioning, APF
+controller periodically populates current concurrency shares for each priority
+level (i.e. concurrency share that can be borrowed or need to be borrowed) by
+taking past usage into consideration. So in combination of those two aspects, a
+priority level's Nomial Share is only an initial value and the actual share
+might increase or decrease according to actual usage with borrowing. However,
+the owner priority level should have full Nominal Share as its current
+concurrency share if in full utilization.
+
+This KEP proposes the following settings:
+
+| Name  | Proposed       | Proposed | Proposed Borrowing | Proposed Guaranteed |
+:       : Nominal Shares : Lendable : Limit              : Shares              :
+| ----- | -------------: | -------: | -----------------: | ------------------: |
+| event | 5              | 0%       | 100%               | 5                   |
```

Comment:
Don't have a number on hand for a scalability test, but I got 4% APF used from some tests.

Reply:
The aggregation of different test sounds even better - thanks!

Reply:
So following up on it.  Currently, the sum of all shares in default configuration across all PLs is 245, with this change it will be bumped to 250. With 5 shares by default and 100% borrowing limit we get a max of 10.
This effectively mean 10 out of 250, which is exactly 4%. 

If we say that in tests we see 4% used for events across aggregated tests, I believe we need to set the upper bound higher (probably by increasing the borrowing limit?)

@MikeSpreitzer - for your thoughts too

Reply:
I agree with @wojtek-t